torch/hub.py

```diff
@@ -140,15 +140,16 @@ def _get_cache_or_reload(github, force_reload, verbose=True):
     # We don't know the repo name before downloading the zip file
     # and inspect name from it.
     # To check if cached repo exists, we need to normalize folder names.
-    repo_dir = os.path.join(hub_dir, '_'.join([repo_owner, repo_name, branch]))
+    normalized_br = branch.replace('/', '_')
+    repo_dir = os.path.join(hub_dir, '_'.join([repo_owner, repo_name, normalized_br]))
```

Comment:
Not sure if the replace trick here works in a portable way; should we look into e.g. `os.sep` or how other folks handle this normalization?

Reply:
Or maybe we can just hash the branch name if there is no reason why we should keep the original naming around.

Reply:
Good point about windows! ;) Yea I'll switch to os.sep and try to make a test for it. 

We might prefer to keep the branch name so that we can reuse cache, hashing will make the cache not discoverable.

Reply:
Oh actually I just tried, github doesn't allow backslash in branch name, so we are good here. ;)